code/game/objects/structures/misc.dm

```diff
@@ -72,4 +72,46 @@
 		user.forceMove(get_turf(src))
 		log_admin("[key_name(user)] was incarnated by a respawner machine.")
 		message_admins("[key_name_admin(user)] was incarnated by a respawner machine.")
-		user.incarnate_ghost()
\ No newline at end of file
+		user.incarnate_ghost()
+
+/obj/structure/ghost_beacon
+	name = "ethereal beacon"
+	desc = "A structure that draws ethereal attention when active. Use an empty hand to activate."
+	icon = 'icons/obj/lavaland/artefacts.dmi'
+	icon_state = "anomaly_crystal"
+	anchored = 1
+	density = 1
+	var/active = FALSE
+	var/ghost_alert_delay = 30 SECONDS
+	var/last_ghost_alert
+
+
+/obj/structure/ghost_beacon/initialize()
+	. = ..()
+	last_ghost_alert = world.time
+	if(active)
+		processing_objects.Add(src)
+
+/obj/structure/ghost_beacon/Destroy()
+	if(active)
+		processing_objects.Remove(src)
+	. = ..()
```

Comment:
I know this is the same thing as `return ..()`, but it's a bit inconsistent to use this at this location.